provisioning_templates/provision/autoyast_default.erb

```diff
@@ -22,21 +22,57 @@ oses:
   <networking>
     <dns>
       <hostname><%= @host.name %></hostname>
+      <% if @host.subnet.respond_to?(:dhcp_boot_mode?) -%>
+      <dhcp_hostname config:type="boolean">false</dhcp_hostname>
+      <dhcp_resolv config:type="boolean">false</dhcp_resolv>
+      <domain><%= @host.domain %></domain>
+      <% if @host.subnet.dns_primary.present? -%>
+      <nameservers config:type="list">
+        <nameserver><%= @host.subnet.dns_primary %></nameserver>
+        <% if @host.subnet.dns_secondary.present? -%>
+        <nameserver><%= @host.subnet.dns_secondary %></nameserver>
+        <% end -%>
+      </nameservers>
+      <% end -%>
+      <% else -%>
       <resolv_conf_policy>auto</resolv_conf_policy>
-<% if @host.domain -%>
+      <% end -%>
+    <% if @host.domain -%>
       <searchlist config:type="list">
         <search><%= @host.domain %></search>
       </searchlist>
-<% end -%>
+    <% end -%>
     </dns>
     <interfaces config:type="list">
       <interface>
-        <bootproto>dhcp</bootproto>
         <device>eth0</device>
+        <ethtool_options>autoneg on</ethtool_options>
+        <% if @host.subnet.respond_to?(:dhcp_boot_mode?) -%>
```

Comment:
If it responds to boot mode but boot mode is set to DHCP then you will still end up with static networking.